### ISSUE TITLE
Choosing an epoch

### ISSUE THREAD

**Issue by martinthomson** [huitema/quic-1wd#5]
Would it be acceptable to define any point in time as time 0?  I think that it would be good, as long as the number of microseconds is consistently calculated as the time since the time that is chosen.

The draft is already fairly imprecise about the definition of the time stamp:

> The time stamp encodes the number of microseconds since the beginning of the connection

Being more deliberate about this imprecision might be good, otherwise people might be inclined to ask how "the beginning of the connection" is defined.

Another alternative is to state that the first TIME_STAMP frame (I think that TIMESTAMP is more correct) that is sent contains a value of 0.  That would be very precise, but it would likely mean tracking that point in time specially, which implementations are unlikely to want to do.

**Comment by huitema** [comment]
There are correctness requirements and performance requirements. The correctness requirement is really about having a single epoch at each end, so peers can track the variations of the time stamp values. The performance requirements are to try use fewer bits for encoding the stamp; starting at zero "at the beginning of the connection" kind of achieve that.

**Comment by martinthomson** [comment]
Right, that's my core point: It would be equally OK to start at the time of the first ACK, or the first packet sent, or the HANDSHAKE_DONE. Whatever event is convenient.  As beginning of the connection is imprecise, people will want to know what you mean.

You could say "any epoch is OK, but times closer to when the first TIME_STAMP is sent will be more efficient." and pre-empt the question, that's all.
